ui/src/containers/configs/NewConfig.js

```diff
@@ -131,6 +154,30 @@ class NewConfig extends Component {
     });
   }
 
+  submitEditorContent() {
+    try {
+      let parsedConfig = JSON.parse(this.state.editorConfig, null, 2);
+      this.props.addDeployment(parsedConfig).then(() => {
+        if (this.props.configs.errorMessage !== undefined) {
+          this.setState({
+            configCreated: false,
+            errorMessage: this.props.config.errorMessage,
+          });
+        } else {
+          this.setState({
+            configCreated: true,
+            errorMessage: "",
+          });
+        }
+      });
+    } catch (syntaxError) {
```

Comment:
From my understanding, we are looking for an error that could occur in line 159? If yes, would it make sense to only wrap this line in a `try{}` block?